crates/stdarch-test/src/lib.rs

```diff
@@ -201,3 +201,42 @@ pub fn assert_skip_test_ok(name: &str, missing_features: &[&str]) {
 
 // See comment in `assert-instr-macro` crate for why this exists
 pub static mut _DONT_DEDUP: *const u8 = std::ptr::null();
+
+#[cfg(target_arch = "x86_64")]
+#[inline]
+pub unsafe fn _init_amx() {
```

Comment:
These syscalls are required only on the linux kernel, and you can use the `syscall` crate as a `dev-dependency`. Windows by default enables amx support. Also, this should be in `core_arch/src/x86/test.rs`

Reply:
Since only a few testcases of amx intrinsics require this function and probably not expose to users, so maybe it is too expensive to introduce the syscall crate

Reply:
It's fine to add dev-dependencies. They are only used when testing this crate in CI.